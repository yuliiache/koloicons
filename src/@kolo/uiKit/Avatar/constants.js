export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
