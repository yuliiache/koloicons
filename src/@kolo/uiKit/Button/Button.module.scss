@import "@kolo/design";

.button {
  cursor: pointer;
  font-weight: 500;
  border: none;
  font-size: 0.8rem;
  border-radius: 4px;
  color: $WhiteBG;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  transition: box-shadow 0.2s ease-out;
  line-height: 1.5;
  box-sizing: border-box;
  text-align: center;
  display: flex;
  justify-content: center;

  &.button-small {
    padding: 12px 16px;
  }

  &.button-big {
    min-height: 56px;
    padding: 16px 24px;
  }

  &.button-full-width {
    width: 100%;
    max-width: 100%;
    padding: 12px 24px;
  }

  &.button-primary {
    background: linear-gradient(to bottom, $PrimaryGradientStart, $PrimaryGradientEnd);
    box-shadow: $primaryBoxShadow;

    &:not(:disabled):active {
      background: linear-gradient(to bottom, $PrimaryGradientEnd, $PrimaryGradientStart);
      box-shadow: $primaryBoxShadowActStart, $primaryBoxShadowActEnd;
    }

    &:not(:disabled):focus {
      box-shadow: $primaryBoxShadow, $primaryBoxShadowFocHov, $boxShadowFocEnd;
    }

    &:not(:disabled):hover {
      box-shadow: $primaryBoxShadow, $primaryBoxShadowFocHov;
    }
  }

  &.button-accent {
    background: linear-gradient(to bottom, $AccentGradientStart, $AccentGradientEnd 99%);
    box-shadow: $accentBoxShadow;

    &:not(:disabled):active {
      box-shadow: $primaryBoxShadowActStart, $primaryBoxShadowActEnd;
    }

    &:not(:disabled):focus {
      box-shadow: $primaryBoxShadow, $primaryBoxShadowFocHov, $boxShadowFocEnd;
    }

    &:not(:disabled):hover {
      box-shadow: $accentBoxShadow, $primaryBoxShadowFocHov;
    }
  }

  &.button-basic {
    background: linear-gradient(to bottom, $BasicGradientStart, $BasicGradientEnd);
    box-shadow: $basicBoxShadow;
    color: $GreyFontBtn;

    &:not(:disabled):active {
      box-shadow: $basicBoxShadowAct, $basicBoxShadowActEnd;
    }

    &:not(:disabled):focus {
      box-shadow: $basicBoxShadowFocHovStart, $basicBoxShadowFocMid, $boxShadowFocEnd;
    }

    &:not(:disabled):hover {
      box-shadow: $basicBoxShadowFocHovStart, $basicBoxShadowHovEnd;
    }
  }

  &.button-disabled {
    background: $GreyDisableBgBtn;
    box-shadow: none;
    cursor: default;
    color: $LightGreyFontBtn;
  }

  &-children {
    justify-content: space-between;
  }
}
