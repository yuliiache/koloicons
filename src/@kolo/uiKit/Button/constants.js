export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
