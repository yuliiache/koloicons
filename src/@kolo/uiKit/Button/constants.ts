export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
