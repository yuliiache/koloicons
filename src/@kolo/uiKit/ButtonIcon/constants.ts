export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
