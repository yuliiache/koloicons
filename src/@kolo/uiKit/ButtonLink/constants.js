export const BUTTONLINK_TEST_ID = 'buttonLinkComponent';

