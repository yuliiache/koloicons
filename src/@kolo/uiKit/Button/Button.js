import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import {BUTTON_COLOR, BUTTON_SIZE, BUTTON_TEST_ID, BUTTON_TEXT_TEST_ID, BUTTON_TYPE} from './constants';

import styles from './Button.module.scss';

export const Button = ({
  color = BUTTON_COLOR.PRIMARY,
  size = BUTTON_SIZE.SMALL,
  text,
  isDisabled,
  onClick,
  type = BUTTON_TYPE.BUTTON,
}) => {
  const buttonClasses = classNames(styles.button, styles[`button-${size}`], styles[`button-${color}`], {
    [styles['button-disabled']]: isDisabled,
  });

  return (
    <button
      data-testid={BUTTON_TEST_ID}
      type={type}
      className={buttonClasses}
      onClick={onClick}
      disabled={isDisabled}
    >
      <span data-testid={BUTTON_TEXT_TEST_ID}>{text}</span>
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(BUTTON_SIZE),
  color: PropTypes.oneOf(BUTTON_COLOR),
  text: PropTypes.string.isRequired,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func,
  type: PropTypes.oneOf(BUTTON_TYPE),
};
