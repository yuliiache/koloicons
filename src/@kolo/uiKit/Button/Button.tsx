import classnames from 'classnames';
import {FC, ReactNode} from 'react';

import styles from './Button.module.scss';
import {BUTTON_TEST_ID, BUTTON_TEXT_TEST_ID, ButtonColor, ButtonSize, ButtonType} from './constants';

export interface ButtonProps {
  text: string;
  isDisabled?: boolean;
  onClick?: () => void;
  color?: ButtonColor;
  size?: ButtonSize;
  type?: ButtonType;
  children?: ReactNode;
}

const Button: FC<ButtonProps> = ({
  text,
  onClick,
  children,
  isDisabled = false,
  color = ButtonColor.PRIMARY,
  size = ButtonSize.SMALL,
  type = ButtonType.BUTTON,
}) => {
  const buttonClasses = classnames(styles.button, styles[`button-${size}`], styles[`button-${color}`], {
    [styles['button-disabled']]: isDisabled,
    [styles['button-children']]: children,
  });

  return (
    <button
      data-testid={BUTTON_TEST_ID}
      type={type}
      className={buttonClasses}
      onClick={onClick}
      disabled={isDisabled}
    >
      <span data-testid={BUTTON_TEXT_TEST_ID}>{text}</span>
      {children && <span>{children}</span>}
    </button>
  );
};

export default Button;
