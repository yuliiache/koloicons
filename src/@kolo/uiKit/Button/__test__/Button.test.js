import React from 'react';
import {cleanup, render, waitFor} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import {Button} from '../Button';
import {BUTTON_COLOR, BUTTON_SIZE, BUTTON_TEST_ID} from '../constants';

describe('Button component', () => {
  afterEach(cleanup);

  const commonProps = {
    size: 'small',
    color: 'primary',
    text: 'Common Text',
    isDisabled: false,
    onClick: jest.fn(),
    type: 'button',
  };

  const colors = Object.values(BUTTON_COLOR);
  const sizes = Object.values(BUTTON_SIZE);

  describe('snapshot', () => {
    const testCases = [{description: 'disabled state', props: {isDisabled: true}}];

    colors.forEach((color) => {
      sizes.forEach((size) => {
        const description = `${color} color, ${size} size`;
        const props = {color, size};
        testCases.push({description, props});
      });
    });

    test.each(testCases)('should render %s snapshot', ({description, props}) => {
      const {asFragment} = render(
        <Button
          {...commonProps}
          {...props}
        />
      );

      expect(asFragment()).toMatchSnapshot(description);
    });
  });

  it('should have text', () => {
    const buttonText = 'Text';
    const {getByText} = render(
      <Button
        {...commonProps}
        text={buttonText}
      />
    );
    const buttonTextElement = getByText(buttonText);

    expect(buttonTextElement).toBeInTheDocument();
  });

  it('should call onClick function', async () => {
    const onClick = jest.fn();
    const {getByTestId} = render(
      <Button
        {...commonProps}
        text="OnClick"
        onClick={onClick}
      />
    );
    const buttonElement = getByTestId(BUTTON_TEST_ID);

    userEvent.click(buttonElement);

    await waitFor(() => expect(onClick).toHaveBeenCalled());
  });
});
