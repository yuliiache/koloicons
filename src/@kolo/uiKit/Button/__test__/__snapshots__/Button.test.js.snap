// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot should render { description: 'accent color, big size', props: [Object] } snapshot: accent color, big size 1`] = `
<DocumentFragment>
  <button
    class="button button-big button-accent"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'accent color, full-width size', props: [Object] } snapshot: accent color, full-width size 1`] = `
<DocumentFragment>
  <button
    class="button button-full-width button-accent"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'accent color, small size', props: [Object] } snapshot: accent color, small size 1`] = `
<DocumentFragment>
  <button
    class="button button-small button-accent"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'basic color, big size', props: [Object] } snapshot: basic color, big size 1`] = `
<DocumentFragment>
  <button
    class="button button-big button-basic"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'basic color, full-width size', props: [Object] } snapshot: basic color, full-width size 1`] = `
<DocumentFragment>
  <button
    class="button button-full-width button-basic"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'basic color, small size', props: [Object] } snapshot: basic color, small size 1`] = `
<DocumentFragment>
  <button
    class="button button-small button-basic"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'disabled state', props: [Object] } snapshot: disabled state 1`] = `
<DocumentFragment>
  <button
    class="button button-small button-primary button-disabled"
    data-testid="buttonComponent"
    disabled=""
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'primary color, big size', props: [Object] } snapshot: primary color, big size 1`] = `
<DocumentFragment>
  <button
    class="button button-big button-primary"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'primary color, full-width size', props: [Object] } snapshot: primary color, full-width size 1`] = `
<DocumentFragment>
  <button
    class="button button-full-width button-primary"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component snapshot should render { description: 'primary color, small size', props: [Object] } snapshot: primary color, small size 1`] = `
<DocumentFragment>
  <button
    class="button button-small button-primary"
    data-testid="buttonComponent"
    type="button"
  >
    <span
      data-testid="buttonComponentText"
    >
      Common Text
    </span>
  </button>
</DocumentFragment>
`;
